DOCSP-29577 Link to C++ Filter Page from RQL Page (#2771)
mongodb/docs-realm: source/realm-query-language.txt

.. tabs::

  .. tab:: Flutter SDK
    :tabid: flutter

    - :ref:`Filter & Sort Results - Flutter SDK <flutter-filter-results>`

  .. tab:: Java SDK
    :tabid: java

    - :ref:`Query Engine - Java SDK <java-client-query-engine>`
    - :ref:`Query with Realm Query Language - Java SDK <java-filter-with-realm-query-language>`

  .. tab:: Kotlin SDK
    :tabid: kotlin

    - :ref:`Filter Data - Kotlin SDK <kotlin-filter-data>`

  .. tab:: .NET SDK
    :tabid: dotnet

    - :ref:`Query Engine - .NET SDK <dotnet-client-query-engine>`

  .. tab:: Node.JS SDK
    :tabid: node

    - :ref:`Query Engine - Node.Js SDK <node-client-query-engine>`
    - :ref:`Filter Queries - Node.Js SDK <node-filter-queries>`

  .. tab:: React Native SDK
    :tabid: react-native

    - :ref:`Query Engine - React Native SDK <react-native-client-query-engine>`
    - :ref:`Filter Queries - React Native SDK <react-native-filter-queries>`

  .. tab:: C++ SDK
    :tabid: cpp

>     The full range of RQL expression are not currently supported by the 
>     C++ SDK. For examples querying Realm in the C++ SDK, refer to:

Reviewer: As written, this is awkward grammatically. It has some is/are agreement issues (and a typo). We can sidestep this by rewriting slightly:

```suggestion
    The C++ SDK Alpha implements only a subset of RQL. 
    For examples querying Realm in the C++ SDK, refer to:
```

      - :ref:`Query Engine - React Native SDK <react-native-client-query-engine>`
      - :ref:`Filter Queries - React Native SDK <react-native-filter-queries>`

.. note:: Swift SDK and C++ SDK do not support Realm Query Language

>    The Swift SDK and C++ SDK do not support querying with Realm Query Language.

Reviewer: This isn't strictly true. Swift uses NSPredicate. C++ does not use NSPredicate, but does not yet support the full range of RQL expressions as it is currently still in development. We should probably instead just title this note something generic like:

> "Swift and C++" 

And in the note body say something like: 
> "For examples of querying Realm in the Swift SDK or C++ SDK, refer to:"

And then keep the links below.

Author: Decided to split into two notes, but open to combining!

Reviewer: After seeing the two notes together on the page, it seems like not the right solution. We try to avoid stacking callouts on top of each other, although there are places where it's unavoidable. But I think we can avoid it here.

Maybe the best option is to just change these tabs starting on ln 34 into a nested bullet list? So remove the tabs entirely, and have bullets like:

```
- C++ SDK

  - :ref:`Filter Data - C++ SDK <cpp-filter-data>`

- Flutter

  - :ref:`Filter & Sort Results - Flutter SDK <flutter-filter-results>`
  
- Java SDK

  - :ref:`Query Engine - Java SDK <java-client-query-engine>`
  - :ref:`Query with Realm Query Language - Java SDK <java-filter-with-realm-query-language>` 
```

The spacing may be a little odd - I know you have to leave a blank line between the first-level indent and second-level indent or it messes up the formatting when it renders on the page. But I don't know if you need a blank line between the second level of indent and the first-level indent of the next SDK. You might have to play with this. When we provide a bullet list of SDKs, we like to keep them in alphabetical order by SDK.

I suggest bullets instead of adding a C++ SDK tab because I think we're at the max tab width here without cutting off the tab names. We've had to go to bullet lists in the App Services docs because the tab names were getting cut off. But I don't think we necessarily need to do anything more in this case than just link out to the C++ page - that page already has a note at the top about not supporting the full range of RQL expressions.
